packages/caliper-core/lib/worker/tx-observers/prometheus-tx-observer.js

```diff
@@ -0,0 +1,185 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+const CaliperUtils = require('../../common/utils/caliper-utils');
+const ConfigUtil = require('../../common/config/config-util');
+const TxObserverInterface = require('./tx-observer-interface');
+
+const express = require('express');
+const appServer = express();
+const promClient = require('prom-client');
+const promGcStats = require('prometheus-gc-stats');
+
+const Logger = CaliperUtils.getLogger('prometheus-tx-observer');
+
+/**
+ * Prometheus TX observer used to maintain Prometheus metrics by acting as a scrape target.
+ */
+class PrometheusTxObserver extends TxObserverInterface {
+    /**
+     * Initializes the instance.
+     * @param {object} options The observer configuration object.
+     * @param {MessengerInterface} messenger The worker messenger instance. Not used.
+     * @param {number} workerIndex The 0-based index of the worker node.
+     */
+    constructor(options, messenger, workerIndex) {
+        super(messenger, workerIndex);
+        this.metricPath = options.metricPath || '/metrics';
+        this.port = Number(options.port) || ConfigUtil.get(ConfigUtil.keys.Monitor.PrometheusScrapePort);
```

Comment:
Is this an inbound port or outbound port ? would be good if the variable was a bit more descriptive